Fix `scriptlevel` multipler (font-size)

Ahmad-S792 opened the pull request:
#### 618be985f329c157b17c026c39fc67fa47f8849b
<pre>
Fix `scriptlevel` multipler (font-size)
<a href="https://bugs.webkit.org/show_bug.cgi?id=281220">https://bugs.webkit.org/show_bug.cgi?id=281220</a>
<a href="https://rdar.apple.com/137671252">rdar://137671252</a>

Reviewed by Frédéric Wang.

The following is not defined in MathML Core specification [1] because
it has `font-size: math`, which WebKit don&apos;t implement yet and this
works as fallback (when the font has no math table).

[1] <a href="https://w3c.github.io/mathml-core/#user-agent-stylesheet">https://w3c.github.io/mathml-core/#user-agent-stylesheet</a>

This patch is to align with old specification [2] of `scriptsizemultiplier`,
which was incorrect in our current implementation.

[2] <a href="https://www.w3.org/TR/REC-MathML/chap3_3.html">https://www.w3.org/TR/REC-MathML/chap3_3.html</a> (Refer: Attributes accepted by &lt;mstyle&gt;)

It is also referred as `scriptPercentScaleDown` and `scriptScriptPercentScaleDown` as
Layout Constants in MathML Core [3].

[3] <a href="https://w3c.github.io/mathml-core/#layout-constants-mathconstants">https://w3c.github.io/mathml-core/#layout-constants-mathconstants</a>

* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/css-styling/width-height-001-expected.txt:
* LayoutTests/mathml/presentation/scripts-subsup.html:
* LayoutTests/mathml/presentation/scripts-underover.html:
* LayoutTests/mathml/presentation/underover-nonstretchy-or-vertical.html:
* LayoutTests/platform/glib/accessibility/math-multiscript-attributes-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/mathml/relations/css-styling/scriptlevel-001-expected.txt:
* LayoutTests/platform/glib/mathml/opentype/horizontal-expected.txt:
* LayoutTests/platform/glib/mathml/opentype/horizontal-munderover-expected.txt:
* LayoutTests/platform/glib/mathml/opentype/opentype-stretchy-horizontal-expected.txt:
* LayoutTests/platform/glib/mathml/presentation/bug159513-expected.txt:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/mathml/relations/css-styling/scriptlevel-001-expected.txt:
* LayoutTests/platform/ios/mathml/opentype/opentype-stretchy-horizontal-expected.txt:
* LayoutTests/platform/mac/accessibility/math-multiscript-attributes-expected.txt:
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/mathml/relations/css-styling/scriptlevel-001-expected.txt:
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/mathml/relations/css-styling/size-containment-001.tentative-expected.txt:
* LayoutTests/platform/mac/mathml/opentype/opentype-stretchy-horizontal-expected.txt:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/mathml/relations/css-styling/size-containment-001.tentative-expected.txt
* Source/WebCore/css/mathml.css:
(msub &gt; * + *, msup &gt; * + *, msubsup &gt; * + *, mmultiscripts &gt; * + *, munder &gt; * + *, mover &gt; * + *, munderover &gt; * + *):
(mroot &gt; *:last-child):

Canonical link: <a href="https://commits.webkit.org/285069@main">https://commits.webkit.org/285069@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8aa3ce828f08884abe3ef70814fbcd19903d6f39

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/71467 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/50880 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/24240 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/75578 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/22672 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/73582 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/58680 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/22491 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/56441 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/14911 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/74533 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/46172 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/61563 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/36890 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/42835 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/19028 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/21013 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/64735 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/19394 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/77297 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/15701 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/18567 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/64154 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/15744 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/61603 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/64149 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/15797 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/12301 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/5938 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/46680 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/1459 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/47751 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/49035 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/47493 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

fred-wang reviewed (approved):
Looks like you need another iteration to update the expectations but otherwise LGTM.

Ahmad-S792 committed:
Fix `scriptlevel` multipler (font-size)
https://bugs.webkit.org/show_bug.cgi?id=281220
rdar://137671252

Reviewed by Frédéric Wang.

The following is not defined in MathML Core specification [1] because
it has `font-size: math`, which WebKit don't implement yet and this
works as fallback (when the font has no math table).

[1] https://w3c.github.io/mathml-core/#user-agent-stylesheet

This patch is to align with old specification [2] of `scriptsizemultiplier`,
which was incorrect in our current implementation.

[2] https://www.w3.org/TR/REC-MathML/chap3_3.html (Refer: Attributes accepted by <mstyle>)

It is also referred as `scriptPercentScaleDown` and `scriptScriptPercentScaleDown` as
Layout Constants in MathML Core [3].

[3] https://w3c.github.io/mathml-core/#layout-constants-mathconstants

* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/css-styling/width-height-001-expected.txt:
* LayoutTests/mathml/presentation/scripts-subsup.html:
* LayoutTests/mathml/presentation/scripts-underover.html:
* LayoutTests/mathml/presentation/underover-nonstretchy-or-vertical.html:
* LayoutTests/platform/glib/accessibility/math-multiscript-attributes-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/mathml/relations/css-styling/scriptlevel-001-expected.txt:
* LayoutTests/platform/glib/mathml/opentype/horizontal-expected.txt:
* LayoutTests/platform/glib/mathml/opentype/horizontal-munderover-expected.txt:
* LayoutTests/platform/glib/mathml/opentype/opentype-stretchy-horizontal-expected.txt:
* LayoutTests/platform/glib/mathml/presentation/bug159513-expected.txt:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/mathml/relations/css-styling/scriptlevel-001-expected.txt:
* LayoutTests/platform/ios/mathml/opentype/opentype-stretchy-horizontal-expected.txt:
* LayoutTests/platform/mac/accessibility/math-multiscript-attributes-expected.txt:
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/mathml/relations/css-styling/scriptlevel-001-expected.txt:
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/mathml/relations/css-styling/size-containment-001.tentative-expected.txt:
* LayoutTests/platform/mac/mathml/opentype/opentype-stretchy-horizontal-expected.txt:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/mathml/relations/css-styling/size-containment-001.tentative-expected.txt
* Source/WebCore/css/mathml.css:
(msub > * + *, msup > * + *, msubsup > * + *, mmultiscripts > * + *, munder > * + *, mover > * + *, munderover > * + *):
(mroot > *:last-child):

Canonical link: https://commits.webkit.org/285069@main
